### PR TITLE
Adds Biggest cruise & Carrier stats commands

### LINKED ISSUE #95: Add command to show how many cruises the carrier has been on
Add a command that searches the current booze cruise database and the historical one, counting how often the carrier has been there

### LINKED ISSUE #292: Add taxi command
Source destination departure 
Post to announcements?

### LINKED ISSUE #342: Biggest cruise command
Request: command to return booze_tally for the historic cruise with the most carrier trips/wine

### LINKED ISSUE #392: New /booze_tally_all_time command
A new command /booze_tally_all_time that would show the tally stats but include carriers from the current cruise as well as all the historical ones

### PR DESCRIPTION
**Pull request by Conshmea**
Add command to return biggest historical cruise tally
Closes #392
Closes #342

Add command to return stats of specific carrier
Closes #95

Changes help command to use role ids not strings for names

### CODE REVIEW

File: ptn/boozebot/botcommands/DatabaseInteraction.py [hunk 1]
```diff
@@ -2111,3 +2124,106 @@ def check_yes_no(check_message):
...
+
+        :param interaction discord.Interaction: The discord interaction context.
+        :param bool extended: If the extended stats should be shown.
+        :returns: None"
+        """
+   
+        print(f"{interaction.user.name} requested the biggest cruise tally, extended: {extended}.")
+        await interaction.response.defer()
+        
+        pirate_steve_lock.acquire()
```
Comment (IndorilReborn) [inline ptn/boozebot/botcommands/DatabaseInteraction.py:2143]: Is there a reason to lock on selects?

Comment (IndorilReborn) [inline ptn/boozebot/botcommands/DatabaseInteraction.py:2143]: Is there a reason to lock on selects?

File: ptn/boozebot/botcommands/DatabaseInteraction.py [hunk 2]
```diff
@@ -529,6 +531,91 @@ def build_stat_embed(
...
+            f"Wine Boxes per capita (🏴󠁧󠁢󠁳󠁣󠁴󠁿): {wine_boxes_per_scot_pop:,.2f}\n\n"
+            f"Olympic Swimming Pool Volume (L): {olympic_swimming_pool_volume:,}\n"
+            f"Olympic Swimming Pools if Bottles of Wine: {pools_if_bottles:,.2f}\n"
+            f"Olympic Swimming Pools if Boxes of Wine: {pools_if_boxes:,.2f}\n\n"
+            f"London Bus Volume (L): {london_bus_volume_l:,}\n"
+            f"London Busses if Bottles of Wine: {busses_if_bottles:,.2f}\n"
+            f"London Busses if Boxes of Wine: {busses_if_boxes:,.2f}\n\n",
+        )
+        stat_embed.set_footer(
+            text="Stats requested by RandomGazz.\nPirate Steve approves of these stats!"
```
Comment (IndorilReborn) [inline ptn/boozebot/botcommands/DatabaseInteraction.py:615]: question: Do we want randomgazz here, or dynamic requester?

File: ptn/boozebot/botcommands/DatabaseInteraction.py [hunk 3]
```diff
@@ -2111,3 +2124,106 @@ def check_yes_no(check_message):
...
+
+        :param interaction discord.Interaction: The discord interaction context.
+        :param str carrier_id: he XXX-XXX carrier ID.
+        :returns: None"
+        """
+        
+        print(f"{interaction.user.name} requested the stats for the carrier: {carrier_id}.")
+        await interaction.response.defer()
+        
+        pirate_steve_lock.acquire()
```
Comment (IndorilReborn) [inline ptn/boozebot/botcommands/DatabaseInteraction.py:2191]: Same as above, Is locking needed?

**Review by IndorilReborn — APPROVED** [review]
LGTM, only nitpicks

File: ptn/boozebot/botcommands/DatabaseInteraction.py (hunk 3 above)
Comment (Conshmea) [reply ptn/boozebot/botcommands/DatabaseInteraction.py:2191]: Nope, idk why I put that there

**Pull request merged**